scripts/signedbuildskadi.sh

```diff
@@ -1,274 +1,311 @@
-echo "  - /opt/skadi_credentials (only if run-time generated credentials chosen)"
-echo "  - /opt/Skadi/Docker/.env"
-echo "  - /opt/Skadi/Docker/skadi_dockprom/.env"
+#!/bin/bash
+set -e
+
+hello_message () {
+  echo "Welcome to installing a secure dockerized container setup of Skadi"
+  echo "Please ensure you have at least 8 GB RAM and 4 cores allocated to the host"
+  read -n 1 -r -s -p "If you already have this configured press any key to continue... or CTRL+C to exit (nothing has been installed)"
+  echo ""
+}
+
+setup_host () {
+  # Update
+  sudo apt-get update && sudo apt-get dist-upgrade -y
+
+  # Install deps
+  sudo apt-get install -y \
+    openssh-server \
+    git \
+    curl \
+    glances \
+    unzip \
+    vim \
+    htop \
+    screen \
+    gnupg \
+    net-tools \
+    software-properties-common \
+    apache2-utils
+
+  # Set the vm.max_map_count kernel setting needs to be set to at least 262144 for production use
+    sudo sysctl -w vm.max_map_count=262144
+    echo vm.max_map_count=262144 | sudo tee -a /etc/sysctl.conf
+
+  # Set Hostname to skadi by default with option to not to
+    if [ $hostname_change = "true" ]
+      then
+      echo "Renaming Host to skadi"
+      newhostname='skadi'
+      oldhostname=$(</etc/hostname)
+      sudo hostname $newhostname >/dev/null 2>&1
+      sudo sed -i "s/$oldhostname/$newhostname/g" /etc/hosts >/dev/null 2>&1
+      echo skadi |sudo tee /etc/hostname >/dev/null 2>&1
+      sudo systemctl restart systemd-logind.service >/dev/null 2>&1
+    else
+      echo "Not renaming host"
+    fi
+
+    # Set Server to UTC or not
+    if [ $set_time_utc = "true" ]
+      then
+      # Set Timezone to UTC
+      sudo timedatectl set-timezone UTC
+    fi
+
+  # Create Skadi user by default with option to not to
+  if [ $create_skadi_user = "true" ]
+    then
+    if ! id -u $SKADI_USER >/dev/null 2>&1; then
+        echo "==> Creating $SKADI_USER user"
+        echo "" >> /opt/skadi_credentials
+        echo "  Created OS Account:" >> /opt/skadi_credentials
+        echo "     - Username: $SKADI_USER" >> /opt/skadi_credentials
+        echo "     - Password: $SKADI_PASS" >> /opt/skadi_credentials
+        /usr/sbin/groupadd $SKADI_USER
+        /usr/sbin/useradd $SKADI_USER -g $SKADI_USER -G sudo -d $SKADI_USER_HOME --create-home -s "/bin/bash"
+        echo "$SKADI_USER:$SKADI_PASS" | chpasswd
+
+        # Set up sudo
+        echo "==> Giving $SKADI_USER sudo powers"
+        echo "$SKADI_USER        ALL=(ALL)       NOPASSWD: ALL" > /etc/sudoers.d/$SKADI_USER
+        chmod 440 /etc/sudoers.d/$SKADI_USER
+    fi
+  else
+    echo "Not creating skadi user"
+    SKADI_USER=$(whoami)
+  fi
+
+  # Enable and Configure UFW Firewall
+  echo "Enabling UFW firewall to only allow OpenSSH and Ngninx Full"
+  sudo ufw allow 80
+  sudo ufw allow 443
+  sudo ufw allow 'OpenSSH'
+  sudo ufw --force enable
+}
+
+host_default_values () {

```

Comment:
Does this work? I thought bash required the variables to be at the top of the file or it would throw errors while parsing the functions.

Reply:
this has been moved to to the top of the file, and removed as a function